tools/ksvalidator.py

```diff
@@ -43,68 +43,81 @@ def cleanup(dest, fn=None, exitval=1):
         except Exception:
             pass
 
-    sys.exit(exitval)
-
-op = argparse.ArgumentParser(usage="%(prog)s [options] ksfile")
-op.add_argument("ksfile", nargs="?",
-                help=_("filename or URL to read from"))
-op.add_argument("-e", "--firsterror", dest="firsterror", action="store_true",
-                default=False, help=_("halt after the first error or warning"))
-op.add_argument("-i", "--followincludes", dest="followincludes",
-                action="store_true", default=False,
-                help=_("parse include files when %%include is seen"))
-op.add_argument("-l", "--listversions", dest="listversions", action="store_true",
-                default=False,
-                help=_("list the available versions of kickstart syntax"))
-op.add_argument("-v", "--version", dest="version", default=DEVEL,
-                help=_("version of kickstart syntax to validate against"))
-
-opts = op.parse_args(sys.argv[1:])
-
-if opts.listversions:
-    for key in sorted(versionMap.keys()):
-        print(key)
-
-    sys.exit(1)
-
-if not opts.ksfile:
-    op.print_usage()
-    sys.exit(1)
-
-destdir = tempfile.mkdtemp("", "ksvalidator-tmp-", "/tmp")
-try:
-    f = load_to_file(opts.ksfile, "%s/ks.cfg" % destdir)
-except KickstartError as e:
-    print(_("Error reading %(filename)s:\n%(version)s") % {"filename": opts.ksfile, "version": e})
-    cleanup(destdir)
-
-try:
-    handler = makeVersion(opts.version)
-except KickstartVersionError:
-    print(_("The version %s is not supported by pykickstart") % opts.version)
-    cleanup(destdir)
-
-ksparser = KickstartParser(handler, followIncludes=opts.followincludes,
-                           errorsAreFatal=opts.firsterror)
-
-# turn DeprecationWarnings into errors
-warnings.filterwarnings("error")
-
-processedFile = None
-
-try:
-    processedFile = preprocessKickstart(f)
-    ksparser.readKickstart(processedFile)
-    cleanup(destdir, processedFile, exitval=0)
-except DeprecationWarning as msg:
-    print(_("File uses a deprecated option or command.\n%s") % msg)
-    cleanup(destdir, processedFile)
-except KickstartParseError as msg:
-    print(msg)
-    cleanup(destdir, processedFile)
-except KickstartError:
-    print(_("General kickstart error in input file"))
-    cleanup(destdir, processedFile)
-except Exception as e:
-    print(_("General error in input file:  %s") % e)
-    cleanup(destdir, processedFile)
+    return exitval
+
+def main(argv=sys.argv[1:]):
+    op = argparse.ArgumentParser(usage="%(prog)s [options] ksfile", add_help=False)
```

Comment:
Sorry but I don't really get you comment. Do you mean to move the argument parser out of the main function so it can be documented automatically ? If this is the case I'll look into it separately b/c I have to continue my work on the kickstart commands documentation anyway.

For this PR I think it can stay as is.